connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -209,7 +209,11 @@ private Long ensureLeaderConfig(long maxOffset) {
         Map<String, List<String>> connectorAssignments = new HashMap<>();
         Map<String, List<ConnectorTaskId>> taskAssignments = new HashMap<>();
 
-        // Perform round-robin task assignment
+        // Perform round-robin task assignment. Assign all connectors and then all tasks because not assigning both
+        // connector and its tasks can lead to very uneven distribution of work in some common cases (e.g. for connectors
+        // that generate only 1 task each; in a cluster of 2 or an even # of nodes, only even nodes will be assigned
+        // connectors and only odd nodes will be assigned tasks, but tasks are, on average, actually more resource
+        // intensive than connectors.
         CircularIterator<String> memberIt = new CircularIterator<>(sorted(allConfigs.keySet()));
```

Comment:
nit: not from this patch, but the name `allConfigs` threw me off a little. Maybe something like `workerConfigs` or `memberConfigs` would be clearer?